aiven/kafka_topic_availability.go

```diff
@@ -71,19 +73,57 @@ func (w *KafkaTopicAvailabilityWaiter) RefreshFunc() resource.StateRefreshFunc {
 }
 
 func (w *KafkaTopicAvailabilityWaiter) refresh() error {
-	if !kafkaTopicAvailabilitySem.TryAcquire(5) {
+	if !kafkaTopicAvailabilitySem.TryAcquire(1) {
 		log.Printf("[TRACE] Kafka Topic Availability cache refresh already in progress ...")
 		return nil
 	}
 	defer kafkaTopicAvailabilitySem.Release(1)
 
-	topic, err := w.Client.KafkaTopics.Get(w.Project, w.ServiceName, w.TopicName)
-	if err != nil {
-		return err
+	c := cache.GetTopicCache()
+
+	// warming up cache if queue is empty
+	warmingUpCacheOne.Do(func() {
+		log.Printf("[DEBUG] Kafka Topic queue is empty, warming up cache!")
+		topics, err := w.Client.KafkaTopics.List(w.Project, w.ServiceName)
+		if err == nil {
+			for _, t := range topics {
+				c.AddToQueue(w.Project, w.ServiceName, t.TopicName)
+			}
+		}
+	})
+
+	if _, ok := c.LoadByTopicName(w.Project, w.ServiceName, w.TopicName); ok {
+		return nil
+	}
+
+	for {
+		queue := c.GetQueue(w.Project, w.ServiceName)
+
+		if len(queue) == 0 {
+			break
+		}
+
+		log.Printf("[DEBUG] Kafka Topic queue: %+v", queue)
+		v2Topics, err := w.Client.KafkaTopics.V2List(w.Project, w.ServiceName, queue)
```

Comment:
Probably makes sense to do in batches with at most, say, 200 items fetched in one go. There will likely be problems if you try getting 10k or however many topics some environment might have in one go.

Reply:
Sorry, didn't read far enough. Looks like it already does this.